### PR TITLE
Updated so that bad phone numbers no longer stop the message from bei…

### PR DESCRIPTION
**Pull request by JessicaWNava**
…ng sent.

This PR allows messages to be sent even if the phone number is bad. 

Test Steps:
1. Update the phone number in prime-router/src/testIntegration/resources/datatests/FHIR_to_HL7/sample_truncate_CA_20230406-0001.fhir to 1 (000)000-0000
2. Run ./prime fhirdata --input-file src/testIntegration/resources/datatests/FHIR_to_HL7/sample_truncate_CA_20230406-0001.fhir -s metadata/hl7_mapping/receivers/STLTs/CA/CA.yml --output-format HL7 --output-file src/testIntegration/resources/datatests/FHIR_to_HL7/testhl7.hl7
3. Make sure it runs without error

## Changes
- try/catch added to allow messages to be sent even if the phone number is bad. 
- tests updated

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

## Linked Issues
- #9644

### CODE REVIEW

File: prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/PhoneUtilities.kt [hunk 1]
```diff
@@ -29,13 +29,15 @@ object PhoneUtilities {
...
-                PhonePart.Extension -> phone.extension
+            return try {
+                when (part) {
+                    PhonePart.Country -> phone.countryCode.toString()
+                    PhonePart.AreaCode -> phone.nationalNumber.toString().substring(0, 3)
+                    PhonePart.Local -> phone.nationalNumber.toString().substring(3)
+                    PhonePart.Extension -> phone.extension
+                }
+            } catch (e: StringIndexOutOfBoundsException) {
+                null
```
Comment (jack-h-wang) [inline prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/PhoneUtilities.kt:40]: Should a warning be logged here per the acceptance criteria?

**Review by jack-h-wang — COMMENTED** [review]
Looks good to me, but I had a comment just to make sure all the acceptance criteria were met!

**Comment by JessicaWNava** [comment]
Yep, updated. I left it as a generic message since I am unsure if adding the phone number would be violating PII rules. Normally it would be but if it is a bad phone number I am unsure. Back to you @jack-h-wang thank you for the quick review!

**Review by jack-h-wang — APPROVED** [review]
Looks good to me!

**Comment by JessicaWNava** [comment]
@arnejduranovic over to you!

**Comment by JessicaWNava** [comment]
> Looks good.
> 
> Please make sure to link the relevant ticket in the PR, will approve once that's done.

I thought we weren't supposed to do that because it would auto mark it as done once it was merged?

**Comment by arnejduranovic** [comment]
> > Looks good.
> > Please make sure to link the relevant ticket in the PR, will approve once that's done.
> 
> I thought we weren't supposed to do that because it would auto mark it as done once it was merged?

We're just not supposed to put "Fixes" in front, that's the magic that closes the ticket. For example, if you just put the ticket number by itself in the section titled "linked issues" or whatever it wont auto close the ticket, like so:

#XXXX

**Comment by JessicaWNava** [comment]
> 

Got it, done.

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**